loadtest/user/userentity/websocket.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package userentity
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
```

Comment:
I'd like to avoid logging in this package. All information should go through the upper layer and eventually become a `control.UserStatus`. Since we are mainly interested in errors we could probably return a `<-chan error` on `Connect()` and use that in the controller to handle related errors.

Reply:
That's much better. But I'm wondering how will we handle that in the controller though. As of now, each action function returns a status, which is channeled up to the controller running it. But now if user.Connect also returns a channel, then I'm not sure how are we going to take the error from the channel and return it ? Because each action function is synchronous.

Somehow we would need to pass the status chan passed to the `(*Controller).Run` to this action method. And then keep sending the statuses to that channel. But that needs slight tinkering with the `run` function signature, which is `run       func() control.UserStatus`.

What do you think ?

Reply:
One option could be to create a goroutine that pipes those errors into the `UserStatus` channel. Something like:

```go
ec := c.user.Connect()
go func() {
  for err := range ec {
    c.status <- c.newErrorStatus(err)
  }
}()
```

Making sure we `close(ec)` on disconnect.

Reply:
Sure, that's fine. Then we need to have the status channel as a field of the controller.

Reply:
Correct. It's definitely not the most clever solution but it's simple and should be good enough for our current needs.